deploy/getsf

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash -e
+
+function on_exit {
+    echo
+    echo "*** Installation failed ***"
+    echo
+    }
+trap 'on_exit $?' EXIT
+
+# Warn people
+echo "This script will make irreversable changes to this system."
+echo "This includes installing packages, changing your networking,"
+echo "and configuring the kernel. Are you sure you want to"
+echo "proceed?"
+echo ""
+echo -n "(yes or no) >> "
+read warning
+if [ ${warning} != "yes" ]; then
+    exit 1
+fi
+echo
+
+# Ask which release track we should be on
+echo "Do you want released or pre-release packages? Generally we"
+echo "recommend released packages unless you have a good reason"
+echo "to need a specific bug fix."
+echo ""
+echo -n "(release or pre-release) >> "
+read release
+pipextra=""
+if [ ${release} == "release" ]; then
+    echo "We will use released packages."
+elif [ ${release} == "pre-release" ]; then
+    echo "We will use pre-release packages."
+    pipextra="--pre"
+else
+    echo "Could not parse response."
+    exit 1
+fi
+echo
+
+# We must run as root to do the things we do.
+if [ $(whoami) != "root" ]; then
+    echo "You need to run this script as root."
+fi
+
+# Are we on Ubuntu >= 20.04 or Debian >= 10?
+osok=0
+isdebian=0
+isubuntu=0
+if [ -e /etc/os-release ]; then
+    source /etc/os-release
+    if [ ${ID} == "debian" ]; then
+	echo "Detected Debian."
+	isdebian=1
+
+	if [ ${VERSION_ID} -gt 9 ]; then
+	    osok=1
+	fi
+    elif [ ${ID} == "ubuntu" ]; then
+	echo "Detected Ubuntu."
+	isubuntu=1
+
+	MAJOR_VERSION=$(echo ${VERSION_ID} | cut -f 1 -d ".")
+	if [ ${MAJOR_VERSION} -gt 19 ]; then
+	    osok=1
+	fi
+    else
+        echo "Distribution ${ID} is unknown."
+    fi
+fi
+
+if [ ${osok} != 1 ]; then
+    echo "Sorry, this does not look like a supported Linux distribution."
+    echo "We currently support Ubuntu 20.04 and onwards, and Debian 10"
+    echo "onwards."
+    exit 1
+fi
+echo
+
+# Install required packages
+echo "Updating package list."
+apt-get update
+echo
+
+if [ ${isubuntu} == 1 ]; then
+    echo "Adding a PPA to provide a more modern Ansible on Ubuntu."
+    apt-get -y install software-properties-common
+    add-apt-repository --yes --update ppa:ansible/ansible
+    apt-get update
+    echo
+fi
+
+echo "Upgrading packages."
+apt-get -y dist-upgrade
+echo
+
+echo "Installing required packages."
+apt-get -y install ansible git tox build-essential python3-dev python3-wheel \
+    python3-pip python3-venv
+echo
+
+echo "Creating python venv."
+mkdir -p /srv/shakenfist/venv
+python3 -mvenv --system-site-packages /srv/shakenfist/venv
+echo
+
+echo "Installing python packages."
+/srv/shakenfist/venv/bin/pip install -U ${pipextra} shakenfist shakenfist_client
```

Comment:
Do you need `--pre` ?

Reply:
`--pre` is hidden in `pipextra` if you request the pre-release track on line 28...